### PR TITLE
add PipApp implementation

### PR DESCRIPTION
**Pull request by yorkie**
This implements #33 by adding a new package `@pipcook/app` which compiles the script source and outputs pipelines and models. And the current script version has the following restrictions:

- single-file only.
- top-level declaration for `learnable` blocks.

A quick start to show how PipApp looks like:

```ts
import { createLearnable, nlp, types } from '@pipcook/app';
const isCooking = createLearnable(async function isCookingImpl(sentence: string) {
  return (await nlp.classify(sentence));
});

(async () => {
  console.log(await isCooking('test'));
})();
```

The pull request TODO:

- [x] generates the pipeline configs.
- [x] creates pipelines and save to the current directory.
- [x] runs configured pipelines.
- [x] builds the executable and complete program.
- [x] documentation

### CODE REVIEW

**Comment by yorkie** [comment]
Complete the saving pipelines.

File: packages/app/src/apis/executable.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+if (pathExistsSync(APP_MANIFEST)) {
+  manifest = readJSONSync(APP_MANIFEST);
+  if (manifest.executable === true) {
+    executable = true;
+  }
+}
+
+function dynamicModelExports(module: string, exports: any) {
+  manifest?.pipelines.filter((pipeline: any) => {
```
Comment (rickycao-qy) [inline packages/app/src/apis/executable.ts:17]: Is it possible to remove any type ~~

File: packages/app/src/compiler/pipelines/vision-classify.base.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
+  "plugins": {
+    "dataCollect": {
+      "package": "./packages/plugins/data-collect/image-classification-data-collect",
+      "params": {
+        "url": "<TODO>"
```
Comment (rickycao-qy) [inline packages/app/src/compiler/pipelines/vision-classify.base.json:6]: nlp-classify.baes.json has the message 'input your CSV zip' and here the message is 'TODO'. This seems not consistent

File: packages/costa/src/runtime.ts [hunk 3]
```diff
@@ -14,10 +14,15 @@ import {
   CondaConfig
 } from './index';
 
-import request from 'request-promise';
+import { get } from 'request-promise';
 import Debug from 'debug';
 
 const debug = Debug('costa.runtime');
+const UserAgent = [
```
Comment (rickycao-qy) [inline packages/costa/src/runtime.ts:21]: Why do we want to include user-agent to identify browser here

Comment (yorkie) [reply packages/costa/src/runtime.ts:21]: We can remove it now :)

**Comment by yorkie** [comment]
@rickycao-qy resolved all nits :)

**Comment by rickycao-qy** [comment]
LGTM

**Pull request merged**